fails when building tag

https://github.com/evryfs/elasticsearch-docker/runs/401570560

```
un eshork/gitleaks-action@v1.0.01s
##[error]Docker run failed with exit code 2
Run eshork/gitleaks-action@v1.0.0
/usr/bin/docker run --name dfbeda250217d8043cdbd1374b7e7bbb0aa_a280fc --label 488dfb --workdir /github/workspace --rm -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/elasticsearch-docker/elasticsearch-docker":"/github/workspace" 488dfb:eda250217d8043cdbd1374b7e7bbb0aa
ERRO[2020-01-21T19:36:20Z] could not find branch refs/tags/7.5.2        
##[error]Docker run failed with exit code 2
```

Also getting the same error in case of branches during a pull request.

https://github.com/Ouranosinc/Magpie/pull/254/checks?check_run_id=417660466

```
Run eshork/gitleaks-action@v1.0.0
/usr/bin/docker run --name dfbd2aa9bcaea9c4b8c996a192f431b1028_78fa17 --label 488dfb --workdir /github/workspace --rm -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e GITHUB_ACTIONS=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/Magpie/Magpie":"/github/workspace" 488dfb:d2aa9bcaea9c4b8c996a192f431b1028
ERRO[2020-01-30T17:11:31Z] could not find branch refs/heads/coverage    
##[error]Docker run failed with exit code 2
```

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -30,7 +30,7 @@ then
 else
     # pushed tag or single commit reference
     # run only from current to master instead of full history
-    GITHUB_REF_MASTER=$(git show master --pretty=format:"%H")
+    GITHUB_REF_MASTER=$(git show origin/master --pretty='format:"%H"')
     gitleaks -v --exclude-forks --redact --threads=1 \
       --branch=$GITHUB_REF_ID \
       --commit-to=$GITHUB_REF_MASTER \

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -4,6 +4,7 @@ set -e # Abort script at first error
 set -u # nounset - Attempt to use undefined variable outputs error message, and forces an exit
 # set -x # verbose (expands commands)
 
+REF_NAME="$(echo $GITHUB_REF | cut -d '/' -f3)"
 gitleaks -v --exclude-forks --redact --threads=1 \
-  --branch=$GITHUB_SHA \
+  --branch=$REF_NAME \
   --repo-path=$GITHUB_WORKSPACE \

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -10,12 +10,12 @@ then
     gitleaks -v --exclude-forks --redact --threads=1 \
       --commit-from=$GITHUB_SHA \
       --commit-to=$GITHUB_HEAD_REF \
-      --repo-path=$GITHUB_WORKSPACE \
+      --repo-path=$GITHUB_WORKSPACE
 else
     # tag or branch name in the form "refs/tags/<ref>"
     REF_NAME="$(echo $GITHUB_REF | cut -d '/' -f3)"
     gitleaks -v --exclude-forks --redact --threads=1 \
       --branch=$REF_NAME \
-      --repo-path=$GITHUB_WORKSPACE \
+      --repo-path=$GITHUB_WORKSPACE
 fi
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -13,27 +13,31 @@ echo "GITHUB_BASE_REF:   ${GITHUB_BASE_REF}"
 cd "$GITHUB_WORKSPACE"
 git fetch --quiet
 
-# branch/tag name in the form "refs/<ref-type>/<ref-id>[/<ref-subtype>]"
-# ref-type: heads|pull|tags
-GITHUB_REF_ID="$(echo $GITHUB_REF | cut -d '/' -f3)"
 if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]] 
 then
-    GITHUB_REF_PULL_TYPE="$(echo $GITHUB_REF | cut -d '/' -f4)"
-    if [[ "${GITHUB_REF_PULL_TYPE}" == "merge" ]]
-    then
-        # during merge, the head ref is the PR branch name
-        GITHUB_REF_ID=${GITHUB_HEAD_REF}
-    fi
     gitleaks -v --exclude-forks --redact --threads=1 \
-      --branch=$GITHUB_REF_ID \
+      --branch=$GITHUB_HEAD_REF \
+      --commit-to=$GITHUB_BASE_REF \
       --repo-path=$GITHUB_WORKSPACE
 else
-    # pushed tag or single commit reference
+    # branch/tag name in the form "refs/<ref-type>/<ref-id>[/<ref-subtype>]"
+    # ref-type: heads|pull|tags
+    GITHUB_REF_NAME="$(echo $GITHUB_REF | cut -d '/' -f3)"
     # run only from current to master instead of full history
-    GITHUB_REF_MASTER=$(git show origin/master --pretty='format:"%H"')
-    gitleaks -v --exclude-forks --redact --threads=1 \
-      --branch=$GITHUB_REF_ID \
-      --commit-to=$GITHUB_REF_MASTER \
-      --repo-path=$GITHUB_WORKSPACE
+    GITHUB_REF_MASTER=$(git show 'origin/master' --pretty='format:%H')
+    
+    if [[ "${GITHUB_REF_NAME}" == "master" ]] 
+    then
+        # push to master (eg: for tagged versioning), only scan the last commit
+        gitleaks -v --exclude-forks --redact --threads=1 \
+          --branch=$GITHUB_REF_NAME \
+          --depth=1 \
+          --repo-path=$GITHUB_WORKSPACE
+    else        
+        gitleaks -v --exclude-forks --redact --threads=1 \
+          --branch=$GITHUB_REF_NAME \
+          --commit-to=$GITHUB_REF_MASTER \
+          --repo-path=$GITHUB_WORKSPACE
+    fi
 fi
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -10,6 +10,8 @@ echo "GITHUB_REF:        ${GITHUB_REF}"
 echo "GITHUB_HEAD_REF:   ${GITHUB_HEAD_REF}"
 echo "GITHUB_BASE_REF:   ${GITHUB_BASE_REF}"
 
+cd "$GITHUB_WORKSPACE" && git fetch --quiet
+
 if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]] 
 then
     # branch name in the form "refs/heads/<ref-id>[/<ref-type>]"

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -4,7 +4,15 @@ set -e # Abort script at first error
 set -u # nounset - Attempt to use undefined variable outputs error message, and forces an exit
 # set -x # verbose (expands commands)
 
-REF_NAME="$(echo $GITHUB_REF | cut -d '/' -f3)"
+if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]] 
+then
+    # last commit on the PR branch
+    REF_NAME=$GITHUB_SHA
+else
+    # tag name in the form "refs/tags/<tag>"
+    REF_NAME="$(echo $GITHUB_REF | cut -d '/' -f3)"
+fi
 gitleaks -v --exclude-forks --redact --threads=1 \
   --branch=$REF_NAME \
   --repo-path=$GITHUB_WORKSPACE \
+

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -6,13 +6,16 @@ set -u # nounset - Attempt to use undefined variable outputs error message, and 
 
 if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]] 
 then
-    # last commit on the PR branch
-    REF_NAME=$GITHUB_SHA
+    # from last commit on the PR to source 
+    gitleaks -v --exclude-forks --redact --threads=1 \
+      --commit-from=$GITHUB_SHA \
+      --commit-to=$GITHUB_HEAD_REF \
+      --repo-path=$GITHUB_WORKSPACE \
 else
-    # tag name in the form "refs/tags/<tag>"
+    # tag or branch name in the form "refs/tags/<ref>"
     REF_NAME="$(echo $GITHUB_REF | cut -d '/' -f3)"
+    gitleaks -v --exclude-forks --redact --threads=1 \
+      --branch=$REF_NAME \
+      --repo-path=$GITHUB_WORKSPACE \
 fi
-gitleaks -v --exclude-forks --redact --threads=1 \
-  --branch=$REF_NAME \
-  --repo-path=$GITHUB_WORKSPACE \
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -10,7 +10,8 @@ echo "GITHUB_REF:        ${GITHUB_REF}"
 echo "GITHUB_HEAD_REF:   ${GITHUB_HEAD_REF}"
 echo "GITHUB_BASE_REF:   ${GITHUB_BASE_REF}"
 
-cd "$GITHUB_WORKSPACE" && git fetch --quiet
+cd "$GITHUB_WORKSPACE"
+git fetch --quiet
 
 # branch/tag name in the form "refs/<ref-type>/<ref-id>[/<ref-subtype>]"
 # ref-type: heads|pull|tags

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -4,18 +4,23 @@ set -e # Abort script at first error
 set -u # nounset - Attempt to use undefined variable outputs error message, and forces an exit
 # set -x # verbose (expands commands)
 
+echo "GITHUB_EVENT_NAME: ${GITHUB_EVENT_NAME}"
+echo "GITHUB_SHA:        ${GITHUB_SHA}"
+echo "GITHUB_REF:        ${GITHUB_REF}"
+echo "GITHUB_HEAD_REF:   ${GITHUB_HEAD_REF}"
+echo "GITHUB_BASE_REF:   ${GITHUB_BASE_REF}"
+
 if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]] 
 then
-    # from last commit on the PR to source 
-    gitleaks -v --exclude-forks --redact --threads=1 \
-      --commit-from=$GITHUB_SHA \
-      --commit-to=$GITHUB_HEAD_REF \
-      --repo-path=$GITHUB_WORKSPACE
-else
-    # tag or branch name in the form "refs/tags/<ref>"
+    # branch name in the form "refs/heads/<ref>"
     REF_NAME="$(echo $GITHUB_REF | cut -d '/' -f3)"
     gitleaks -v --exclude-forks --redact --threads=1 \
       --branch=$REF_NAME \
+      --repo-path=$GITHUB_WORKSPACE
+else
+    # pushed tag or single commit reference
+    gitleaks -v --exclude-forks --redact --threads=1 \
+      --commit=$GITHUB_SHA \ 
       --repo-path=$GITHUB_WORKSPACE
 fi
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -5,5 +5,5 @@ set -u # nounset - Attempt to use undefined variable outputs error message, and 
 # set -x # verbose (expands commands)
 
 gitleaks -v --exclude-forks --redact --threads=1 \
-  --branch=$GITHUB_REF \
+  --branch=$GITHUB_SHA \
   --repo-path=$GITHUB_WORKSPACE \

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -12,10 +12,11 @@ echo "GITHUB_BASE_REF:   ${GITHUB_BASE_REF}"
 
 cd "$GITHUB_WORKSPACE" && git fetch --quiet
 
+# branch/tag name in the form "refs/<ref-type>/<ref-id>[/<ref-subtype>]"
+# ref-type: heads|pull|tags
+GITHUB_REF_ID="$(echo $GITHUB_REF | cut -d '/' -f3)"
 if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]] 
 then
-    # branch name in the form "refs/heads/<ref-id>[/<ref-type>]"
-    GITHUB_REF_ID="$(echo $GITHUB_REF | cut -d '/' -f3)"
     GITHUB_REF_PULL_TYPE="$(echo $GITHUB_REF | cut -d '/' -f4)"
     if [[ "${GITHUB_REF_PULL_TYPE}" == "merge" ]]
     then
@@ -27,8 +28,11 @@ then
       --repo-path=$GITHUB_WORKSPACE
 else
     # pushed tag or single commit reference
+    # run only from current to master instead of full history
+    GITHUB_REF_MASTER=$(git show master --pretty=format:"%H")
     gitleaks -v --exclude-forks --redact --threads=1 \
-      --commit=$GITHUB_SHA \
+      --branch=$GITHUB_REF_ID \
+      --commit-to=$GITHUB_REF_MASTER \
       --repo-path=$GITHUB_WORKSPACE
 fi
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -20,7 +20,7 @@ then
 else
     # pushed tag or single commit reference
     gitleaks -v --exclude-forks --redact --threads=1 \
-      --commit=$GITHUB_SHA \ 
+      --commit=$GITHUB_SHA \
       --repo-path=$GITHUB_WORKSPACE
 fi
 

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -12,10 +12,16 @@ echo "GITHUB_BASE_REF:   ${GITHUB_BASE_REF}"
 
 if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]] 
 then
-    # branch name in the form "refs/heads/<ref>"
-    REF_NAME="$(echo $GITHUB_REF | cut -d '/' -f3)"
+    # branch name in the form "refs/heads/<ref-id>[/<ref-type>]"
+    GITHUB_REF_ID="$(echo $GITHUB_REF | cut -d '/' -f3)"
+    GITHUB_REF_PULL_TYPE="$(echo $GITHUB_REF | cut -d '/' -f4)"
+    if [[ "${GITHUB_REF_PULL_TYPE}" == "merge" ]]
+    then
+        # during merge, the head ref is the PR branch name
+        GITHUB_REF_ID=${GITHUB_HEAD_REF}
+    fi
     gitleaks -v --exclude-forks --redact --threads=1 \
-      --branch=$REF_NAME \
+      --branch=$GITHUB_REF_ID \
       --repo-path=$GITHUB_WORKSPACE
 else
     # pushed tag or single commit reference

